### DISCUSSION TITLE
conventions for shared plugin configuration

### DISCUSSION THREAD

**Discussion by stefanprobst** [unifiedjs/unified#102] [Q&A] [answered]
hi, are there conventions which unified plugins should accept configuration via `processor.use({ settings })` or `processor.data({ settings })`. it look like parser/compiler do but transformers don't - is this correct?

**Comment by wooorm** [comment, marked as answer]
My advice will depend on what you’re building. Without that, I’d say:
- Do not care about `.data` if you’re making your own plugins: `options` should work fine
- There is also `file.data`, which has some use in certain cases
- settings are somewhat of a legacy, not needed a lot anymore
